draco/analysis/transform.py

```diff
@@ -1265,3 +1265,235 @@ def process_finish(self) -> Union[containers.SiderealStream, containers.RingMap]
+                red_ax_id = current_ax
+
+            # If we can't use the current axis here, walk through until
+            # we find the highest priority axis that we can use
+            if sel_ax_id is None:
+                for ax in ax_priority:
+                    if ax not in options:
+                        continue
+                    if ax not in self.selections.keys():
+                        # Set the first axis to distribute across
+                        sel_ax_id = options.index(ax)
+                        # If possible, use this axis again
+                        if ax not in self.axes and red_ax_id is None:
+                            red_ax_id = sel_ax_id
+                        break
+
+            # If we stil haven't found a good axis for the second step,
+            # walk through the priority tree again until we find one
+            if red_ax_id is None:
+                for ax in ax_priority:
+                    if ax not in options:
+                        continue
+
+                    if ax not in self.axes:
+                        red_ax_id = options.index(ax)
+                        break
+
+            # If we couldn't find a valid axis for at least one step,
+            # throw an exception
+            if sel_ax_id is None or red_ax_id is None:
+                raise ValueError("Could not find an axis to distribute across.")
+
+            return sel_ax_id, red_ax_id
+
+        def _apply_sel(arr, sel, axis):
+            """Apply a selection to a single axis of an array."""
+            if type(sel) is slice:
+                sel = (slice(None),) * axis + (sel,)
+                return arr[sel]
+            elif type(sel) in {list, tuple}:
+                return np.take(arr, sel, axis=axis)
+
+        # Figure out the axes for the new container
+        # Apply the downselections to each axis index_map
+        output_axes = {
+            ax: _apply_sel(data.index_map[ax], sel, 0)
+            for ax, sel in self.selections.items()
+        }
+        # Further reduce any axis indices that will be reduced
+        reduced_axes = {ax: np.array([-1], dtype=int) for ax in self.axes}
+        # Overwrite any axes which will be reduced after selections are made
+        output_axes.update(reduced_axes)
+
+        # Create the output container. Add some extra metadata about which axes
+        # were reduced and which datasets are meaningful
+        out = data.__class__(copy_from=data, **output_axes)
+        out.attrs["reduced"] = True
+        out.attrs["reduction_op"] = self.op
+        out.attrs["reduction_axes"] = np.array(self.axes)
+        out.attrs["downselections"] = self.selections
+        out.attrs["reduced_datasets"] = np.array(self.datasets)
+
+        # Iterate over the datasets and apply operation
+        for name, ds in data.datasets.items():
+            if name not in self.datasets:
+                # Just set this dataset to be zero if it was initialized
```

Comment:
I'm confused about this block: I would have thought that the `copy_from=data` option when `out` was created implies that datasets will be copied from `data` to `out`, so if there's a dataset that's in `out` but not in `self.datasets`, we'd want to leave it alone since it's not being reduced. Am I misunderstanding the logic?

Reply:
I think I may understand now: if there's a dataset that is not being reduced, we don't care about preserving it in the reduced container, so we just set its entries to zero. Is that right? If so, it would be helpful to say that in the comment

Reply:
My idea was exactly what you said, if we aren't reducing a dataset then we don't care about preserving it, but in that case it probably just shouldn't be initialized in the first place. I think I can tweak the code a bit to make that more obvious, and add a comment + docstring entry.

Reply:
Got it, thanks!